src/opa.js

```diff
@@ -1,8 +1,48 @@
 // Copyright 2018 The OPA Authors.  All rights reserved.
 // Use of this source code is governed by an Apache2
 // license that can be found in the LICENSE file.
-const builtIns = require("./builtins");
-
+const builtIns = require("./builtins/index");
+
+/**
+ * @typedef {number & { __type: 'ctx_addr'}} ContextAddress
+ */
+
+/**
+ * @typedef {number & { __type: 'value_addr'}} ValueAddress
+ */
+
+/**
+ * @typedef {number & { __type: 'value_addr'}} StringAddress
+ */
+
+/**
+ * @typedef {ContextAddress | ValueAddress | StringAddress} Address
+ */
+
+/**
+ * @typedef {Object} OpaExports
+ * @property {(ctx_addr: ContextAddress) => number} eval
+ * @property {() => ValueAddress} builtins
+ * @property {() => ContextAddress} opa_eval_ctx_new
+ * @property {(ctx_addr: ContextAddress, value_addr: ValueAddress) => void} opa_eval_ctx_set_input
+ * @property {(ctx_addr: ContextAddress, value_addr: ValueAddress) => void} opa_eval_ctx_set_data
+ * @property {(ctx_addr: ContextAddress) => ValueAddress} opa_eval_ctx_get_result
+ * @property {(size: number) => Address} opa_malloc
+ * @property {(str_addr: StringAddress, size: number) => ValueAddress} opa_json_parse
+ * @property {(value_addr: ValueAddress) => StringAddress} opa_json_dump
+ * @property {(addr: Address) => void} opa_heap_ptr_set
+ * @property {() => Address} opa_heap_ptr_get
+ * @property {(addr: Address) => void} opa_heap_top_set
+ * @property {() => Address} opa_heap_top_get
+ */
```

Comment:
If I'm understanding correctly these are defining the exported fields from the WASM binary, if they are not exposed via API's in the module do we need typescript definitions for them?

On the assumption we do need/want them, can you add some comments describing what these are for how/when to update them, etc.

Reply:
It's mostly for the benefit of type checking code inside the module. I'll go ahead and remove this, would make more sense to keep if ya'll were going all in on typescript.

Reply:
Ah, ok, that makes sense.

For now lets keep them out, but I could see it being useful to have this kind of exported information available _somewhere_ for the OPA WASM binary (more than just in the OPA docs). We can revisit this later on as needed.